### PR TITLE
Fix/sass tilde import

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | internal MANAGER-6999
| License          | BSD 3-Clause

## Description

See : https://webpack.js.org/loaders/sass-loader/#resolving-import-at-rules

using `~` in sass import rules is deprecated

### CODE REVIEW

File: packages/manager/tools/component-rollup-config/src/index.ts [hunk 1]
```diff
@@ -52,6 +61,19 @@ const generateConfig = (opts, pluginsOpts) =>
         sass({
           insert: true,
           output: false,
+          options: {
+            importer(url) {
```
Comment (frenautvh) [inline packages/manager/tools/component-rollup-config/src/index.ts:65]: in order to handle sass imports correctly in rollup i had to manually resolve url using node resolver
(see: https://rollupjs.org/guide/en/#why-isnt-node-resolve-a-built-in-feature)

File: packages/manager/apps/telecom/src/app/app-scss.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@import '~bootstrap4/scss/functions';
```
Comment (frenautvh) [inline packages/manager/apps/telecom/src/app/app-scss.scss:1]: those paths were wrong

**Review by Jisay — APPROVED** [review]

**Review by zakariasahmane — DISMISSED** [review]

**Review by cbourgois — APPROVED** [review]

**Comment by antleblanc** [comment]
@frenautvh Looks like there is a mismatch between versions for both `webpack`
and `sass-loader` from here (according to your description).

Here we are still using `webpack@4`, so we should rely on following
documentation: https://v4.webpack.js.org/loaders/sass-loader/#resolving-import-at-rules

As you can see, it also reports the warning about `~` usage but if we pay
attention to the `sass-loader` changelog, this warning only appears since
`sass-loader@11.0.0`.
See: https://github.com/webpack-contrib/sass-loader/releases/tag/v11.0.0

According to the `yarn.lock` file, here we are still using `sass-loader@8.0.2`
so I'm wondering if following changes make sense or if some addition
dependencies requires to be upgraded too.

Please let me know your thought about it.
Thanks

**Comment by frenautvh** [comment]
> @frenautvh Looks like there is a mismatch between versions for both `webpack`
> and `sass-loader` from here (according to your description).
> 
> Here we are still using `webpack@4`, so we should rely on following
> documentation: https://v4.webpack.js.org/loaders/sass-loader/#resolving-import-at-rules
> 
> As you can see, it also reports the warning about `~` usage but if we pay
> attention to the `sass-loader` changelog, this warning only appears since
> `sass-loader@11.0.0`.
> See: https://github.com/webpack-contrib/sass-loader/releases/tag/v11.0.0
> 
> According to the `yarn.lock` file, here we are still using `sass-loader@8.0.2`
> so I'm wondering if following changes make sense or if some addition
> dependencies requires to be upgraded too.
> 
> Please let me know your thought about it.
> Thanks

I think it would make easier to upgrade later or switch to another bundler by not using the `~` syntax.
From sass-loader webpack point of view i see that removing it has no impact,

**Comment by antleblanc** [comment]
As you are updating 70%+ Sass imports, are you considering, for consistency,
to "import" Sass files with (or without) `.scss` extension?

I'm still having trouble to understand why we are applying changes "encouraged"
by `sass-loader@11.0.0` whereas we keep using `sass-loader@8.0.2`.

According to Sass language documentation:
> The Sass team discourages the continued use of the @import rule..

see: https://sass-lang.com/documentation/at-rules/import

**Comment by frenautvh** [comment]
> As you are updating 70%+ Sass imports, are you considering, for consistency,
> to "import" Sass files with (or without) `.scss` extension?

Yes we could use the same syntax everywhere ; it depends if we explicitely want to specify that we are importing sass or not. Apart from that it doesn't make any difference since both syntax are valid.

> I'm still having trouble to understand why we are applying changes "encouraged"
> by `sass-loader@11.0.0` whereas we keep using `sass-loader@8.0.2`.

The '~' import syntax is supported by sass-loader from webpack but let's say we are using another bundler instead of webpack. Then we may have trouble importing existing sass styles since we are using the '~' syntax introduced by sass-loader. It's better to make our sass code not dependant on sass-loader.

File: packages/manager/modules/pci/src/projects/project/workflow/index.scss [hunk 3]
```diff
@@ -1,9 +1,9 @@
 ovh-manager-kubernetes-list {
-  @import '~bootstrap4/scss/_functions';
-  @import '~bootstrap4/scss/_variables';
-  @import '~bootstrap4/scss/_mixins';
-  @import '~bootstrap4/scss/_utilities';
-  @import '~bootstrap4/scss/_grid';
+  @import 'bootstrap4/scss/_functions';
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/workflow/index.scss:2]: It looks like we are importing bootstrap4 a lot, maybe we can import it at a parent level (out of scope, just a thought)

**Review by marie-j — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/workflow/index.scss (hunk 3 above)
Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/workflow/index.scss:2]: Yes we might have something to do here, i also found that scoping bootstrap might look like it's a best practice but in reality it makes the css dist a LOTS heavier : each bootstrap's class definition is duplicated for the new scope.

**Review by JeremyDec — DISMISSED** [review]

**Review by Heiiji — DISMISSED** [review]

**Comment by frenautvh** [comment]
rebased on develop 0a30f6da7109b85aa33bb8c2adfdf47831eaae94

**Review by marie-j — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by Heiiji — APPROVED** [review]

**Pull request merged**